py/server/modules/snapshot_methods.py

```diff
@@ -0,0 +1,145 @@
+import datetime
+import re
+import string
+import base58
+from .definitions import MOCK_MC_ADDRESS_MAP, mock_nsc, MOCK_OWNER_SC_ADDR_LIST
+from .nsc_methods import get_nsc_ownerships, get_nsc_owner_sc_addresses
+from .proposal import VotingProposal
+from .util_methods import write_proposal_to_file
+
+active_proposal = VotingProposal(in_id=None)
+
+
+
+def extract_body_attributes(body_string):
+    # "Body": "Start: 18 Apr 23 13:40 UTC, End: 18 Apr 23 13:45 UTC, Author: 0xA0CCf49aDBbdfF7A814C07D1FcBC2b719d674959"
+    s1 = re.split(',', body_string)
+    from_string = re.split('Start: ', s1[0])[1]
+    to_string = re.split('End: ', s1[1])[1]
+    auth_string = re.split('Author: ', s1[2])[1]
+
+    from_time = datetime.datetime.strptime(from_string, '%d %b %y %H:%M %Z')
+    to_time = datetime.datetime.strptime(to_string, '%d %b %y %H:%M %Z')
+
+    return from_time, to_time, auth_string
+
+
+def store_proposal_data(proposal_json, chain_tip_height, chain_tip_hash):
+    global active_proposal
+
+    prop_id = proposal_json['ProposalID']
+    if prop_id in proposal_dict.keys():
+        return
+
+    from_time, to_time, author = extract_body_attributes(body_string=proposal_json['Body'])
+
+    prop = VotingProposal(
+        in_id=prop_id,
+        bl_height=chain_tip_height,
+        bl_hash=chain_tip_hash,
+        from_time=from_time,
+        to_time=to_time,
+        author=author)
+
+    proposal_dict[prop_id] = prop
+    active_proposal = prop
+    write_proposal_to_file(active_proposal)
+
+def init_active_proposal(deserialized_proposal_dict):
```

Comment:
If the balancer goes offline for some time and it restarts it is possible that it loads an outdated proposal and uses it's MC height to answer on a new proposal.
It doesn't solve entirely the issue but I think introducing a check on the expiration of the loaded proposal could be a little improvement.